Align guild gems box with other page elements

keysmashes opened the pull request:
Fixes #7796
### Changes

Aligns right side with speech bubble below, and aligns top with description box to the left. Also centres text and gem icon.

Old:

![image](https://cloud.githubusercontent.com/assets/9433472/17119614/e6f548d2-52be-11e6-968c-c17dc05428a4.png)

New:

![image](https://cloud.githubusercontent.com/assets/9433472/17119648/1946b442-52bf-11e6-98f7-bd405e914229.png)

One side-effect of these changes is that the 'guild gems' text is now underlined on hover. Hopefully that doesn't matter? I've no idea why it happens.

I still need to work on this to fix issues created by this change in other places, like the Fortify modal - applying `display: flex` to align the text vertically causes the green plus box visible on hover to be misplaced on top of the gem count, rather than to the side [solution - `display: inline-flex`]. Also, the '+' icon is underlined, which is a problem [fixed with inline-flex].

---

UUID: `0e1d15f7-f36b-4f15-a9e1-0c31b732dac9`

keysmashes committed:
Align guild gems box with other page elements

Fixes #7796.

keysmashes committed:
Fix issues caused by aligning guild gems box

keysmashes commented:
I fixed places like the Fortify modal so the green box appears in its proper place, and I stopped the text being underlined on hover:

![image](https://cloud.githubusercontent.com/assets/9433472/17382797/9f9a203a-59ca-11e6-9ecd-0039eeaa463d.png)

![image](https://cloud.githubusercontent.com/assets/9433472/17383043/d789155e-59cb-11e6-9d40-7b1ac33d7664.png)

![image](https://cloud.githubusercontent.com/assets/9433472/17383163/677562f8-59cc-11e6-903b-f892af47b887.png)

and on hover:

![image](https://cloud.githubusercontent.com/assets/9433472/17382807/ac92a0fa-59ca-11e6-92bc-668be5044afb.png)

![image](https://cloud.githubusercontent.com/assets/9433472/17383056/dfb9df56-59cb-11e6-8c21-4a1c2f14342e.png)

![image](https://cloud.githubusercontent.com/assets/9433472/17383175/7952bbe2-59cc-11e6-8cc4-f40c7a2c3252.png)

The Purchase Gems modal looks just as it did:

![image](https://cloud.githubusercontent.com/assets/9433472/17382999/a92f4d4a-59cb-11e6-992b-1456a88dd133.png)

and the original point of the PR looks just the same:

![image](https://cloud.githubusercontent.com/assets/9433472/17383105/1474b2ac-59cc-11e6-9484-ccc864b12bcf.png)

khipkin commented on website/client/css/tasks.styl:
```diff
@@ -509,13 +509,15 @@ form
   background-color: lighten($best, 20%)
   &:hover, &:focus
     background-color: lighten($best, 40%)
+// this affects the guild gems box, maybe other stuff?
```
Remove this comment.

keysmashes replied:
Done.

keysmashes committed:
Remove unnecessary comment

crookedneighbor commented:
Personally, I think the rounded gem box with the rounded plus button looks weird. Since we're about to do a redesign of the site, I'm not going to sweat small graphical things like that, so if @Alys and @lemoness are fine with it, I'm fine with merging this.

keysmashes commented:
@crookedneighbor I also wasn't sure about that - I can change it so that both boxes always have squared corners in modals with that green box (or revert to squared corners throughout for consistency), or I can have the corners that touch between the two boxes square off when the green box is visible, which would look something like this:

![image](https://cloud.githubusercontent.com/assets/9433472/17449543/2b5999fa-5b53-11e6-85a9-89490be03db8.png)

I can nudge the gem icon up (really, stop it being pushed down) to align it properly as well:

![image](https://cloud.githubusercontent.com/assets/9433472/17449599/993ca2dc-5b53-11e6-8db1-67442b06c0f7.png)

I think that looks better, but I guess you might not want tons more styling if there's about to be a redesign; it's your call.

crookedneighbor commented:
^ I like that

keysmashes committed:
Connect gem total box and plus/add gems button

When the add gems button is displayed, set border-radius on touching
corners between the two buttons to 0, so that they are displayed as a
unit rather than two separate buttons.

keysmashes commented:
It now looks like this:

![image](https://cloud.githubusercontent.com/assets/9433472/17452521/bd85b02a-5b67-11e6-8926-00f6ca689329.png)

![image](https://cloud.githubusercontent.com/assets/9433472/17452524/c8dfb826-5b67-11e6-86e4-cbe66810c3ca.png)

Other things look the same as they did.

lemoness commented:
I agree that we can revisit any changes post-redesign, so Blade, if you're
comfortable, go ahead and merge.

On Fri, Aug 5, 2016 at 4:08 PM, Josh Holland notifications@github.com
wrote:

> It now looks like this:
> 
> [image: image]
> https://cloud.githubusercontent.com/assets/9433472/17452521/bd85b02a-5b67-11e6-8926-00f6ca689329.png
> 
> [image: image]
> https://cloud.githubusercontent.com/assets/9433472/17452524/c8dfb826-5b67-11e6-86e4-cbe66810c3ca.png
> 
> Other things look the same as they did.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/HabitRPG/habitrpg/pull/7829#issuecomment-237986203,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/ADVQwb7Dm798m7NyUjx5jl87Du7P3MZ3ks5qc8JqgaJpZM4JUnsH
> .

crookedneighbor commented:
Notated toward your next tier!
